### PR TITLE
Update `distributions.rst` to list deprecated distribution classes

### PR DESCRIPTION
**Pull request by c-bata**
## Motivation

To make it easier to understand which distribution classes are deprecated.

## Description of the changes

Split the autosummary section to list deprecated distribution classes.

### CODE REVIEW

**Review by y0z — APPROVED** [review]
LGTM

File: docs/source/reference/distributions.rst [hunk 1]
```diff
@@ -13,12 +13,19 @@ Optuna users should not use distribution classes directly, but instead use utili
 
    FloatDistribution
    IntDistribution
+   CategoricalDistribution
+   distribution_to_json
+   json_to_distribution
+   check_distribution_compatibility
+
+The following modules are deprecated and will be removed in the future.
```
Comment (not522) [inline docs/source/reference/distributions.rst:21]: I think "classes" is more suitable.

```suggestion
The following classes are deprecated and will be removed in the future.
```

Comment (c-bata) [reply docs/source/reference/distributions.rst]: You are correct. Thank you for your review!

**Review by not522 — APPROVED** [review]
LGTM!

**Pull request merged**